### PR TITLE
Clean up and split TensorFlow deps out of util/text.py

### PR DESCRIPTION
**Pull request by reuben**
This PR removes some unused code and splits code that depends on TensorFlow out of `util/text.py` and into `util/ctc.py` so that `from util import text` does not require TensorFlow to be installed. This should help with the work Alex is doing on evaluating the accuracy of a TFLite model.

It also does some small refactoring of the test report generation functions.

### CODE REVIEW

**Comment by lissyx** [comment]
Thanks for that, it'll make #1854 nicer

File: util/text.py [hunk 1]
```diff
@@ -56,74 +54,39 @@ def size(self):
...
+    total_wer = 0.0
+    total_word_length = 0.0
+
+    for original, result in zip(originals, results):
+        total_cer += levenshtein(original, result)
+
+        total_wer += levenshtein(original.split(), result.split())
+        total_word_length += len(original.split())
+
+    return total_wer / total_word_length, total_cer / len(originals)
```
Comment (kdavis-mozilla) [inline util/text.py]: Is the returned CER correct? Shouldn't it be:
```python
return total_wer / total_word_length, total_cer / sum([len(original) for original in originals])
```

Looking a bit more the the code this replaces, this isn't really returning the CER. It's returning what was before called the "mean_edit_distance".

From the old calculations the "mean_edit_distance" seems to be the average CER per word, which is exactly what this function is returning.

So this is fine as is.

File: evaluate.py [hunk 2]
```diff
@@ -67,19 +66,16 @@ def calculate_report(labels, decodings, distances, losses):
     '''
     samples = pmap(process_decode_result, zip(labels, decodings, distances, losses))
 
-    total_levenshtein = sum(s.levenshtein for s in samples)
-    total_label_length = sum(s.label_length for s in samples)
-
-    # Getting the WER from the accumulated levenshteins and lengths
-    samples_wer = total_levenshtein / total_label_length
+    # Getting the WER and CER from the accumulated edit distances and lengths
+    samples_wer, samples_cer = wer_cer_batch(labels, decodings)
```
Comment (kdavis-mozilla) [inline evaluate.py]: Is ``samples_cer`` really the CER of the samples?

Seems like it's more the average CER per word for the samples, which is slightly different from the CER for the samples, see my comments in ```wer_cer_batch()```

Maybe a renaming of ```wer_cer_batch()``` and ```samples_cer``` are in order?

File: util/text.py [hunk 3]
```diff
@@ -56,74 +54,39 @@ def size(self):
...
-    values = tuple[1]
-    results = [''] * tuple[2][0]
-    for i in range(len(indices)):
-        index = indices[i][0]
-        results[index] += alphabet.string_from_label(values[i])
-    # List of strings
-    return results
-
-def wer(original, result):
+def wer_cer_batch(originals, results):
```
Comment (kdavis-mozilla) [inline util/text.py]: Should this really be named `wer_cer_batch()`?

It's not really returning the CER, but something like the average CER per word.

Returning the CER would be something like
```python
return total_wer / total_word_length, total_cer / sum([len(original) for original in originals])
```
but it returns
```python
return total_wer / total_word_length, total_cer / len(originals)
```
which is more the the average CER per word.

File: util/text.py (hunk 1 above)
Comment (reuben) [reply util/text.py]: IIRC it's the average CER over sentences, not words. Does that change your reasoning?

Comment (kdavis-mozilla) [reply util/text.py]: Oh yeah it is average CER over sentences, not words, but it's still not the CER as far as I can see.

Comment (reuben) [reply util/text.py]: I can rename it back to mean_edit_distance or something, although it's one hell of a long name to use in functions etc. Another possibility is to change it to actually compute the CER over the corpus, by dividing by the sum of lengths. I'm biased towards the latter.

Comment (kdavis-mozilla) [reply util/text.py]: I'm biased towards the latter too, as it's far more clear what's going on.

**Pull request merged**